polygon/p2p/service.go

```diff
@@ -81,3 +87,34 @@ func (s *service) Penalize(ctx context.Context, peerId PeerId) error {
 func (s *service) ListPeersMayHaveBlockNum(blockNum uint64) []PeerId {
 	return s.peerTracker.ListPeersMayHaveBlockNum(blockNum)
 }
+
+// TODO(taratorio): move inside MessageListener
```

Comment:
you can tidy this up now after you rebase - as discussed message listener is now accepting observers for the decoded messages